src/common/hooks/DefaultTopicProgress/DefaultTopicProgress.tsx

```diff
@@ -0,0 +1,126 @@
+import { useEffect, useState, useContext, useMemo } from 'react'
+import log from 'loglevel'
+import { useNavigate } from 'react-router-dom'
+import { AuthContext, SnackbarContext } from '@services'
+import { usePersistedStore, useStore } from '@store'
+import {
+  LearningPathElement,
+  LearningPathElementReturn,
+  LearningPathElementStatus,
+  LearningPathElementStatusReturn,
+  Topic,
+  User
+} from '@core'
+import { SnackbarMessageProps } from '@components'
+
+/**
+ * Calculate the progress of the learning elements in a topic
+ * @param allLearningElementsInTopic - All learning elements in a topic
+ * @param allDoneLearningElements - All learning elements that are done
+ * @returns - A tuple with the number of done learning elements and the total number of learning elements
+ */
+const calculateLearningElementProgress = (
+  allLearningElementsInTopic: LearningPathElement,
+  allDoneLearningElements: LearningPathElementStatus[]
+): [number, number] => {
+  const allDoneLearningElementsInTopic = allLearningElementsInTopic.path.map((learningElement) => {
+    return allDoneLearningElements.some((status) => status.cmid === learningElement.learning_element.lms_id)
+  })
+  return [
+    allDoneLearningElementsInTopic.filter((stateDone) => stateDone).length,
+    allLearningElementsInTopic.path.length
+  ]
+}
```

Comment:
Diese Funktion kann auch direkt in den Hook gepackt werden, insofern sie nicht exportiert werden muss.

Reply:
kann aber auch draußen bleiben für bessere übersicht